common/component/kafka/metadata.go

```diff
@@ -324,24 +324,42 @@ func (k *Kafka) getKafkaMetadata(meta map[string]string) (*KafkaMetadata, error)
 	}
 
 	// producer connection specifications
-	if val, ok := meta[clientConnectionRefreshIntervalMin]; ok && val != "" {
-		v, err := strconv.Atoi(val)
+	if val, ok := meta[clientConnectionRefreshInterval]; ok && val != "" {
+		// Parse the duration string
+		dur, err := time.ParseDuration(val)
 		if err != nil {
-			return nil, err
+			// If parsing as duration fails, check if it's a number and interpret as minutes
+			intVal, err := strconv.Atoi(val)
+			if err != nil {
+				return nil, fmt.Errorf("kafka error: invalid value for '%s' attribute: %w", clientConnectionRefreshInterval, err)
+			}
+			dur = time.Duration(intVal) * time.Minute
+		}
+		m.clientConnectionRefreshInterval = dur
+		if m.clientConnectionRefreshInterval <= 0 {
+			m.clientConnectionRefreshInterval = defaultClientConnectionRefreshInterval
 		}
-		m.clientConnectionRefreshIntervalMin = v
 	} else {
-		m.clientConnectionRefreshIntervalMin = defaultClientConnectionRefreshIntervalMin
+		m.clientConnectionRefreshInterval = defaultClientConnectionRefreshInterval
 	}
 
-	if val, ok := meta[clientConnectionMaxKeepAliveMin]; ok && val != "" {
-		v, err := strconv.Atoi(val)
+	if val, ok := meta[clientConnectionKeepAliveInterval]; ok && val != "" {
+		// Parse the duration string
+		dur, err := time.ParseDuration(val)
 		if err != nil {
-			return nil, err
+			// If parsing as duration fails, check if it's a number and interpret as minutes
+			intVal, err := strconv.Atoi(val)
+			if err != nil {
+				return nil, fmt.Errorf("kafka error: invalid value for '%s' attribute: %w", clientConnectionKeepAliveInterval, err)
+			}
+			dur = time.Duration(intVal) * time.Minute
+		}
+		m.clientConnectionKeepAliveInterval = dur
+		if m.clientConnectionKeepAliveInterval < 0 {
+			m.clientConnectionKeepAliveInterval = defaultClientConnectionKeepAliveInterval
 		}
-		m.clientConnectionMaxKeepAliveMin = v
 	} else {
-		m.clientConnectionMaxKeepAliveMin = defaultClientConnectionMaxKeepAliveMin
+		m.clientConnectionKeepAliveInterval = defaultClientConnectionKeepAliveInterval
```

Comment:
All this code is not necessary now that you have set the properties to be of type `time.Duration`. You can just remove all these lines...

Reply:
You need to set the default values in the struct before calling `decodeMetadata`, here: https://github.dev/sicoyle/components-contrib/blob/b572ee936a38cc32ca609fe414247b3c18fd167e/common/component/kafka/metadata.go#L155-L161 Then `decodeMetadata` will overwrite the defaults if there's a value explicitly set in the metadata

Reply:
setting the defaults where you've stated, and then removing these lines of code also fails. While it does take the defaults, if I set a value such as for my test:
```

	t.Run("setting producer values explicitly", func(t *testing.T) {
		k := getKafka()
		m := getCompleteMetadata()
		m[clientConnectionRefreshInterval] = "3m0s"
		m[clientConnectionKeepAliveInterval] = "4m0s"

		meta, err := k.getKafkaMetadata(m)
		require.NoError(t, err)
		require.Equal(t, 3*time.Minute, meta.clientConnectionRefreshInterval)
		require.Equal(t, 4*time.Minute, meta.clientConnectionKeepAliveInterval)
	})
```

Then relying upon the decodeMetadata fails. This to me means there is a bug in the decodeMetadata when it comes to parsing time durations. However, this too falls outside of the scope of the intended work for this PR. I would like that follow Github best practices and fall outside of the realm of this PR to keep the scope in check with what the PR is intended to be for.